![alt text](logo.png "Title")
<center> <h1> Data Science Technical Test <h1></center>

## Instructions
This assessment intends to evaluate your technical skills. The task has intentionally been
designed to be somewhat open ended to allow you to be propositive.
The dataset that must be used for this evaluation is found in the document attached. You
must address the requirements described in the deliverable description. You will also have to
present your work in a meeting

### Data extraction

In [1]:
import pandas as pd

In [15]:
df = pd.read_csv('Dataset for data_scientist test.txt', delimiter = ",")

In [16]:
df.head()

key_1      date_2  cont_3  cont_4  disc_5 disc_6      cat_7   cat_8  \
0      na  2003-04-06   -0.90   -3.82       1     19      happy    blue   
1  UV7055  2013-08-22   -0.35    2.23       5     13  surprised   green   
2  oM1806  2002-10-10    0.93  -12.12       2     11      happy   green   
3  YJ8175  2015-01-01    1.23   -7.21       1     14     scared  purple   
4  VF7543  2014-03-03   -1.87   -9.02       3      9      happy    blue   

   cont_9  cont_10 label  
0    0.37     3.92   2.5  
1    0.05     0.60  3.19  
2    0.08     0.75  2.33  
3   -0.69    -7.24  2.54  
4    1.01    11.16   2.1

In [27]:
def minMax(x):
    return pd.Series(index=['min','max'], data=[x.min(), x.max()])


df.apply(minMax)

key_1      date_2  cont_3  cont_4  disc_5 disc_6      cat_7   cat_8  \
min  AD3812  2000-01-03   -3.22  -31.75       0      0      happy    blue   
max  zq5501  2020-12-31    3.32   33.77       5     na  surprised  yellow   

     cont_9  cont_10 label  
min   -2.99   -35.35  0.99  
max    3.00    35.20    na

In [21]:
df.cat_7.unique()

array(['happy', 'surprised', 'scared', 'sad'], dtype=object)

In [22]:
df.cat_8.unique()

array(['blue', 'green', 'purple', 'orange', 'yellow', 'red'], dtype=object)

# Feature description

In this problem we don't have enough inforamaion about the business problem, so we need to try to find the most information we could from the dataset. Following there are the supostition about the features of th dataset we found.

- key_1: this is the primary key and is alphanumeric
- date_2: this is the date of the record.
- cont_3: this a continue variable with values in a range of -3.22 and 3.22
- cont_4: this a continue variable with values in a range of -31.75 and 33.77
- disc_5: this a discrete variable with values in a range of 0 and 5
- disc_6: this a discrete variable with values in a range of 0 and 21
- cat_7: this a categorical variable with values: 'happy', 'surprised', 'scared', 'sad'
- cat_8: this a categorical variable with values: 'blue', 'green', 'purple', 'orange', 'yellow', 'red'
- cont_9: this a continue variable with values in a range of -2.99 and 3
- cont_10: this a continue variable with values in a range of -35.35 and 35.20
- label: this is the indepent variable in our dataset and what we want to predict

### Data preparation

We need to verify the data types of our features fix it if it isn't correct

In [28]:
df.dtypes

key_1       object
date_2      object
cont_3     float64
cont_4     float64
disc_5       int64
disc_6      object
cat_7       object
cat_8       object
cont_9     float64
cont_10    float64
label       object
dtype: object

date_2, disc_6, and label don't have the correct data type, let's fix this.

### Data transformation

### Model training

### Model prediction

### Recommendations to the analytics team